internal/parse/parser.go

```diff
@@ -278,6 +304,138 @@ func (p *Parser) parseGoFullName() (FullName, bool, error) {
+// objects. Otherwise it returns an empty array, false and an error.
+func (p *Parser) parseList(parseFn func(p *Parser) (FullName, bool, error)) ([]FullName, bool, error) {
+	cp := p.save()
+	if p.skipByte('(') {
+		parenPos := p.pos
+		p.skipSpaces()
+
+		nextItem := true
+		var objs []FullName
+		for nextItem {
+			if obj, ok, err := parseFn(p); ok {
+				objs = append(objs, obj)
+				p.skipSpaces()
+			} else if err != nil {
+				return nil, false, err
+			} else {
+				return nil, false, fmt.Errorf("invalid identifier near char %d", p.pos)
+			}
+			p.skipSpaces()
+			if p.skipByte(')') {
+				return objs, true, nil
+			}
+			nextItem = p.skipByte(',')
+			p.skipSpaces()
+		}
+		return nil, false, fmt.Errorf("missing closing parentheses for char %d", parenPos)
+	}
+	cp.restore()
+	return nil, false, nil
+}
+
+// parseColumns parses text in the SQL query of the form "table.colname". If
+// there is more than one column then the columns must be enclosed in brackets
+// e.g.  "(col1, col2) AS &Person.*".
+func (p *Parser) parseColumns() ([]FullName, bool) {
+	cp := p.save()
+	// We skip a space here to keep consistent with parseTargets which also
+	// consumes one space before the start of the expression.
+	p.skipByte(' ')
+	// Case 1: A single column e.g. p.name
+	if col, ok, _ := p.parseColumn(); ok {
+		return []FullName{col}, true
+	} else if cols, ok, _ := p.parseList((*Parser).parseColumn); ok {
+		return cols, true
+	}
+	cp.restore()
+	return nil, false
+}
+
+// parseTargets parses the part of the output expression following the
+// ampersand. This can be one or more Go objects. If the ampersand is not found
+// or is not preceded by a space and succeeded by a name or opening bracket the
+// targets are not parsed.
+func (p *Parser) parseTargets() ([]FullName, bool, error) {
+	cp := p.save()
+
+	// An '&' must be preceded by a space and succeeded by a name or opening
+	// bracket.
+	if p.skipString(" &") {
+		// Case 1: A single target e.g. &Person.name
+		if target, ok, err := p.parseGoFullName(); ok {
+			return []FullName{target}, true, nil
+		} else if err != nil {
+			return nil, false, err
+			// Case 2: Multiple targets e.g. &(Person.name, Person.id)
+		} else if targets, ok, err := p.parseList((*Parser).parseGoFullName); ok {
+			if starCount(targets) > 1 {
+				return nil, false, fmt.Errorf("more than one asterisk in expression near char %d", p.pos)
+			}
+			return targets, true, nil
+		} else if err != nil {
+			return nil, false, err
+		}
+	}
+	cp.restore()
+	return nil, false, nil
+}
+
+// starCount returns the number of FullNames in the argument with a asterisk in
+// the Name field.
+func starCount(fns []FullName) int {
+	s := 0
+	for _, fn := range fns {
+		if fn.Name == "*" {
+			s++
+		}
+	}
+	return s
+}
+
+// parseOutputExpression parses all output expressions. The ampersand must be
+// preceded by a space and followed by a name byte.
+func (p *Parser) parseOutputExpression() (op *OutputPart, ok bool, err error) {
```

Comment:
If you choose to change `op` to something else, it should be updated here also :)

Reply:
Done.